core/edge/layer_test/src/main.rs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use log::debug;
+use log::info;
+use opendal::Result;
+use opendal::{services, Operator};
+
+use opendal::layers::*;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let builder = services::Memory::default();
+    let registry = prometheus::default_registry();
+
+    let op = Operator::new(builder)
```

Comment:
Adding a new edge test like this doesn't seem ideal. How about we implement something similar to a behavior test for layers in `opendal::layers`? This way, we can run them as unit tests for each layer.

Reply:
Yeah, this would be better.